src/app/pages/bitcoin-contract-request/bitcoin-contract-request.tsx

```diff
@@ -14,55 +17,75 @@ import { BitcoinContractRequestWarningLabel } from './components/bitcoin-contrac
 
 export function BitcoinContractRequest() {
   const getNativeSegwitSigner = useCurrentAccountNativeSegwitSigner();
+  const navigate = useNavigate();
 
-  const { handleOffer, handleAccept, handleReject } = useBitcoinContracts();
+  const { handleOffer, handleAccept, handleReject, sendRpcResponse } = useBitcoinContracts();
 
   const [bitcoinContractJSON, setBitcoinContractJSON] = useState<string>();
   const [bitcoinContractOfferDetails, setBitcoinContractOfferDetails] =
     useState<BitcoinContractOfferDetails>();
   const [bitcoinAddress, setBitcoinAddress] = useState<string>();
+  const [attestorURLs, setAttestorURLs] = useState<string[]>([]);
 
   const [isLoading, setLoading] = useState(true);
 
   const handleAcceptClick = async () => {
     if (!bitcoinContractJSON || !bitcoinContractOfferDetails) return;
 
-    await handleAccept(bitcoinContractJSON, bitcoinContractOfferDetails.counterpartyWalletDetails);
+    await handleAccept(
+      bitcoinContractJSON,
+      bitcoinContractOfferDetails.counterpartyWalletDetails,
+      attestorURLs
+    );
   };
 
   const handleRejectClick = async () => {
     if (!bitcoinContractOfferDetails) return;
 
-    await handleReject(bitcoinContractOfferDetails.simplifiedBitcoinContract.bitcoinContractId);
+    handleReject(bitcoinContractOfferDetails.simplifiedBitcoinContract.bitcoinContractId);
   };
 
   useOnMount(() => {
     const bitcoinContractOfferJSON = initialSearchParams.get('bitcoinContractOffer');
-    const counterpartyWalletURL = initialSearchParams.get('counterpartyWalletURL');
-    const counterpartyWalletName = initialSearchParams.get('counterpartyWalletName');
-    const counterpartyWalletIcon = initialSearchParams.get('counterpartyWalletIcon');
+    const counterpartyWalletDetailsJSON = initialSearchParams.get('counterpartyWalletDetails');
+    const attestorURLs = initialSearchParams.get('attestorURLs');
+
+    const bitcoinAccountDetails = getNativeSegwitSigner?.(0);
+
+    if (!bitcoinAccountDetails) return;
+
+    const currentBitcoinNetwork = bitcoinAccountDetails.network;
+
+    if (currentBitcoinNetwork !== 'testnet') {
+      navigate(RouteUrls.BitcoinContractLockError, {
+        state: {
+          error: new Error('invalid network'),
+          title: 'The account you are using is not on the Bitcoin Testnet',
+          body: 'Unable to interact with Bitcoin Contract',
+        },
+      });
+      sendRpcResponse('none', '', 'failed');
```

Comment:
This should be an `rpcErrorResponse`, rather than a success response with `failed`?